packages/flutter_tools/lib/src/usage.dart

```diff
@@ -19,11 +20,16 @@ const String _kFlutterUA = 'UA-67589403-6';
 Usage get flutterUsage => Usage.instance;
 
 class Usage {
-  /// Create a new Usage instance; [versionOverride] is used for testing.
-  Usage({ String settingsName: 'flutter', String versionOverride }) {
+  /// Create a new Usage instance; [versionOverride] and [configDirOverride] are
+  /// used for testing.
+  Usage({ String settingsName: 'flutter', String versionOverride, String configDirOverride}) {
     final FlutterVersion flutterVersion = FlutterVersion.instance;
     final String version = versionOverride ?? flutterVersion.getVersionString(whitelistBranchName: true);
-    _analytics = new AnalyticsIO(_kFlutterUA, settingsName, version);
+    _analytics = new AnalyticsIO(_kFlutterUA, settingsName, version,
+        // Analyzer doesn't recognize that [Directory] objects match up due to a
+        // conditional import.
+        // ignore: argument_type_not_assignable
```

Comment:
fyi @pq

Reply:
Thanks for the heads-up.  I spoke to Janice a bit and it sounds like this is a known problem with a known solution (at least to @bwilkerson :)) but that it's not trivial.  Some relevant background can be gathered from: https://github.com/dart-lang/sdk/issues/30143 but be warned: there be dragons! 😄